packages/components/src/components/hds/app-footer/index.ts

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+
+import type { ComponentLike } from '@glint/template';
+import type { HdsYieldSignature } from '../yield';
+import type HdsAppFooterItemSignature from './legal-links';
+import type HdsAppFooterLegalLinksSignature from './legal-links';
+import type HdsAppFooterLinkSignature from './link';
+import type HdsAppFooterStatusLinkSignature from './status-link';
+
+export interface HdsAppFooterSignature {
+  Args: {
+    ariaLabel?: string;
+    copyrightYear?: string;
+    theme?: string;
+  };
+  Blocks: {
+    default: [
+      {
+        ExtraBefore?: ComponentLike<HdsYieldSignature>;
+        Item?: ComponentLike<HdsAppFooterItemSignature>;
+        LegalLinks?: ComponentLike<HdsAppFooterLegalLinksSignature>;
+        Link?: ComponentLike<HdsAppFooterLinkSignature>;
+        StatusLink?: ComponentLike<HdsAppFooterStatusLinkSignature>;
+        ExtraAfter?: ComponentLike<HdsYieldSignature>;
```

Comment:
For consistency with [the documentation](https://helios.hashicorp.design/components/app-footer?tab=code#appfooter)
```suggestion
        ExtraBefore?: ComponentLike<HdsYieldSignature>;
        StatusLink?: ComponentLike<HdsAppFooterStatusLinkSignature>;
        LegalLinks?: ComponentLike<HdsAppFooterLegalLinksSignature>;
        Link?: ComponentLike<HdsAppFooterLinkSignature>;
        Item?: ComponentLike<HdsAppFooterItemSignature>;
        ExtraAfter?: ComponentLike<HdsYieldSignature>;
```